src/Features/CSharp/Portable/ConvertPrimaryToRegularConstructor/ConvertPrimaryToRegularConstructorCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,504 @@
+
+                    var constructorDeclaration = CreateConstructorDeclaration().WithAdditionalAnnotations(constructorAnnotation);
+
+                    // If there is an existing non-static constructor, place it before that
+                    var firstConstructorIndex = currentTypeDeclaration.Members.IndexOf(m => m is ConstructorDeclarationSyntax c && !c.Modifiers.Any(SyntaxKind.StaticKeyword));
+                    if (firstConstructorIndex >= 0)
+                    {
+                        return currentTypeDeclaration.WithMembers(
+                            currentTypeDeclaration.Members.Insert(firstConstructorIndex, constructorDeclaration));
+                    }
+
+                    // No constructors.  Place after any fields if present, or any properties if there are no fields.
+                    var lastFieldOrProperty = currentTypeDeclaration.Members.LastIndexOf(m => m is FieldDeclarationSyntax);
+                    if (lastFieldOrProperty < 0)
+                        lastFieldOrProperty = currentTypeDeclaration.Members.LastIndexOf(m => m is PropertyDeclarationSyntax);
+
+                    if (lastFieldOrProperty >= 0)
+                    {
+                        constructorDeclaration = constructorDeclaration
+                            .WithPrependedLeadingTrivia(ElasticCarriageReturnLineFeed);
+
+                        return currentTypeDeclaration.WithMembers(
+                            currentTypeDeclaration.Members.Insert(lastFieldOrProperty + 1, constructorDeclaration));
+                    }
+
+                    // Nothing at all.  Just place the construct at the top of the type.
+                    return currentTypeDeclaration.WithMembers(
+                        currentTypeDeclaration.Members.Insert(0, constructorDeclaration));
+                });
+        }
+
+        async Task RewritePrimaryConstructorParameterReferencesAsync()
+        {
+            foreach (var (parameter, references) in parameterReferences)
+            {
+                // Only have to update references if we're synthesizing a field for this parameter.
+                if (!parameterToSynthesizedFields.TryGetValue(parameter, out var field))
+                    continue;
+
+                var fieldName = field.Name.ToIdentifierName();
+
+                foreach (var grouping in references.GroupBy(r => r.SyntaxTree))
+                {
+                    var syntaxTree = grouping.Key;
+                    var editor = await solutionEditor.GetDocumentEditorAsync(solution.GetDocumentId(syntaxTree), cancellationToken).ConfigureAwait(false);
+
+                    foreach (var identifierName in grouping)
+                    {
+                        var xmlElement = identifierName.AncestorsAndSelf().OfType<XmlEmptyElementSyntax>().FirstOrDefault();
+                        if (xmlElement is { Name.LocalName.ValueText: "paramref" })
```

Comment:
Not sure I understand here .why `paramref` needs to be changed?

Reply:
A type decl's doc comment can refer to the primary constructor parameters using the paramref construct. However, since the type will no longer have parameters on it anymore, these will become errors as they will no longer bind to anything.  They have to be converted to 'see' tags that now point at the generated field.